src/actions/actions.jsx

```diff
@@ -1,5 +1,65 @@
-export function toggleSetting(name) {
-  return { type: 'TOGGLE_SETTING', name };
-}
+import $ from 'jquery';
 
-export default toggleSetting;
+// Account Page Component
+export const toggleSetting = name => ({
+  type: 'TOGGLE_SETTING',
+  name,
+});
+
+// Login Component
+export const updatePassword = password => ({
+  type: 'UPDATE_PASSWORD',
+  password,
+});
+
+export const updateUsername = username => ({
+  type: 'UPDATE_USERNAME',
+  username,
+});
+
+// Register Component
+export const updateNewUsername = username => ({
+  type: 'UPDATE_NEW_USERNAME',
+  username,
+});
+
+export const updateNewPassword = password => ({
+  type: 'UPDATE_NEW_PASSWORD',
+  password,
+});
+
+export const updateConfirmPassword = password => ({
+  type: 'UPDATE_CONFIRM_PASSWORD',
+  password,
+});
+
+// Login backend interaction
+export const receiveUser = json => ({
+  type: 'RECEIVE_USER',
+  username: json.username,
+  id: json.id,
+});
+
+export const login = (username, password) => (
+  (dispatch) => {
+    const data = {
+      username,
+      password,
+    };
+    $.post('/login', data)
+      .done(response => dispatch(receiveUser(response)))
+      .fail((err, status) => console.log(err, status));
```

Comment:
Is there any way we can have an alert error pop up when the user doesn't have an account/password is incorrect?